scripts/globals/abilities/pets/somnolence.lua

```diff
@@ -0,0 +1,48 @@
+---------------------------------------------------
+-- Somnolence
+---------------------------------------------------
+
+require("scripts/globals/settings");
+require("scripts/globals/status");
+require("scripts/globals/monstertpmoves");
+require("scripts/globals/magic");
+
+---------------------------------------------------
+
+function onAbilityCheck(player, target, ability)
+    return 0,0;
+end;
+
+function onPetAbility(target, pet, skill)
+    local spell = getSpell(165); --gave the ability the same power as Thunder II
+    local dmg = calculateMagicDamage(178,1,pet,spell,target,ELEMENTAL_MAGIC_SKILL,MOD_INT,false);
+    local dINT = pet:getStat(MOD_INT) - target:getStat(MOD_INT);
+    local resist = applyPlayerResistance(pet,-1,target, dINT, ELEMENTAL_MAGIC_SKILL, ELE_DARK);
+    local tp = skill:getTP();
+    local duration = 120;
+
+    dmg = dmg*resist;
+    dmg = mobAddBonuses(pet,spell,target,dmg, ELE_DARK);
+
+    if (tp < 100) then
+        tp = 100;
+    end
+
+    dmg = dmg * tp / 100;
```

Comment:
Every pet magical move in trunk does the same thing there.
```lua
if (tp < 100) then
    tp = 100;
end
dmg = dmg * tp / 100;
```

Except I think he didn't need delHP() at all because 
`dmg = finalMagicAdjustments(pet,target,spell,dmg);`
Would actually apply the damage. (I'm basing this assumption on the fact none of the existing scripts do a delHP() and that is their last line)